src/ORM/Query.php

```diff
@@ -37,23 +37,23 @@
  *
  * @see \Cake\Collection\CollectionInterface For a full description of the collection methods supported by this class
  * @method \Cake\Collection\CollectionInterface each(callable $c) Passes each of the query results to the callable
- * @method \Cake\Collection\CollectionInterface sortBy($callback, $dir = SORT_DESC, $type = \SORT_NUMERIC) Sorts the query with the callback
+ * @method \Cake\Collection\CollectionInterface sortBy($callback, int $dir = SORT_DESC, int $type = \SORT_NUMERIC) Sorts the query with the callback
  * @method \Cake\Collection\CollectionInterface filter(callable $c = null) Keeps the results using passing the callable test
  * @method \Cake\Collection\CollectionInterface reject(callable $c) Removes the results passing the callable test
  * @method bool every(callable $c) Returns true if all the results pass the callable test
  * @method bool some(callable $c) Returns true if at least one of the results pass the callable test
  * @method \Cake\Collection\CollectionInterface map(callable $c) Modifies each of the results using the callable
  * @method mixed reduce(callable $c, $zero = null) Folds all the results into a single value using the callable.
  * @method \Cake\Collection\CollectionInterface extract($field) Extracts a single column from each row
- * @method mixed max($field, $type = SORT_NUMERIC) Returns the maximum value for a single column in all the results.
- * @method mixed min($field, $type = SORT_NUMERIC) Returns the minimum value for a single column in all the results.
+ * @method mixed max($field, int $type = SORT_NUMERIC) Returns the maximum value for a single column in all the results.
+ * @method mixed min($field, int $type = SORT_NUMERIC) Returns the minimum value for a single column in all the results.
  * @method \Cake\Collection\CollectionInterface groupBy(string|callable $field) In-memory group all results by the value of a column.
  * @method \Cake\Collection\CollectionInterface indexBy(string|callable $field) Returns the results indexed by the value of a column.
- * @method int countBy(string|callable $field) Returns the number of unique values for a column
+ * @method \Cake\Collection\CollectionInterface countBy(string|callable $field) Returns the number of unique values for a column
```

Comment:
Doesn't countBy return an int?

Reply:
As weird as it might sound: Definitely a collection object
https://github.com/cakephp/cakephp/blob/master/src/Collection/CollectionInterface.php#L501
https://github.com/cakephp/cakephp/blob/master/src/Collection/CollectionTrait.php#L281